apps/web/src/components/FileUploader.tsx

```diff
@@ -78,3 +77,27 @@ export const FileUploader = ({ extensions, handleClose, open, upload }: FileUplo
     </Modal>
   );
 };
+
+function handleFileWithRestrictions(file: File, extensions: string[] = []) {
+  try {
+    // file non-empty
+    // file extensions
+    // Dropzone would make file empty if extension is not correct, so we don't need to check for extension here
+    if (!file) {
+      throw new Error(`Select a file of type: ${extensions.join(', ')}`);
+    }
+
+    // file size < 6MB
+    if (file.size > 6 * 1024 * 1024) {
+      throw new Error('File is too large! Please select a file smaller than 6MB');
+    }
+
+    return file;
+  } catch (err: unknown) {
+    if (err instanceof Error) {
+      toast.warning(err.message);
+    } else {
+      console.error(`An unknown error occurred: ${err}`);
```

Comment:
Can we include a toast message here, just so the user gets a message.

Reply:
good suggestion! thanks